api/controllers/apps_test.go

```diff
@@ -85,14 +96,26 @@ func TestAppCreateWithAlreadyExists(t *testing.T) {
 	test.AssertStatus(t, 403, "POST", "http://convox/apps", val)
 }
 
+func TestAppCreateWithAlreadyExistsBound(t *testing.T) {
+	aws := test.StubAws(
+		test.DescribeStackNotFound("application"),
+		test.CreateAppStackExistsCycle("convox-test-application"),
+		test.DescribeAppStackCycle("convox-test-application"),
+	)
+	defer aws.Close()
+
+	val := url.Values{"name": []string{"application"}}
+	test.AssertStatus(t, 403, "POST", "http://convox/apps", val)
+}
```

Comment:
The request cycles look great here.

It would also make the test clearer to demonstrate the error string doesn't say anything about a legacy app.